### ISSUE TITLE
Unclear on why direct Object relationships (vs. Links) are useful

### ISSUE THREAD

**Issue by strugee** [w3c/activitystreams#364]
Please Indicate One: 
- [ ] Editorial
- [X] Question
- [ ] Feedback
- [ ] Blocking Issue
- [ ] Non-Blocking Issue

Please Describe the Issue:

I'm given to understand that the difference between Links and Objects is an important part of ActivityStreams 2.0. However, while it's obvious why having Links is useful (additional metadata annotations), I can't find anything in the core spec that illustrates why keeping the option to reference Objects is useful.

In particular section 4.2 Link states:

>  Formally, the former example establishes an unqualified direct relationship with the image resource while the latter creates a qualified, indirect relationship that allows additional properties about the relationship to be specified.

Why can't the former example (which IIUC currently is a direct reference to an Object):

```
"image": "http://example.org/application/123.png"
```

simply imply a Link with no additional metadata, such that the above is semantically equivalent to:

```
  "image": {
    "type": "Link",
    "href": "http://example.org/application/123.png"
  }
```

Basically, this distinction just doesn't seem useful to me. I think including an example of a situation where this might be used in the specification text would be super helpful.

**Comment by jasnell** [comment]
They are quite close, yes, and in most cases functionally equivalent but there is an important distinction. Specifically, the reason for `Link` is to allow properties of the *reference* to be described as opposed to properties of the *resource*. Take the following example, for instance:

I have an object that references an image. The publisher of the image includes an English caption, "My cat". In my object referencing the image, however, I want to provide a French translation of that caption, "Mon chat". This alternative title is a property of my reference, not a property of the original picture, so if i did something like:

```js
{
  "image": {
    "id": "http://example.org/application/123.png",
    "type": "Image",
    "title": "Mon chat"
  }
}
```

Then I would be wrong because I would be ascribing the new title property to an object that I do not own. The right approach is to use a `Link`:

```js
{
  "image": {
    "href": "http://example.org/application/123.png",
    "type": "Link",
    "title": "Mon chat"
  }
}
```

Functionally they get you to the same place, but *semantically* they are quite different. If I wanted to load these into a graph database to do semantic queries and analysis, I would end up with significantly different results based on what I'm asking for.

**Comment by strugee** [comment]
@jasnell thanks, that was really helpful. Perhaps consider putting something like that in the spec?

**Comment by jasnell** [comment]
@strugee ... noted! :-)

**Comment by evanp** [comment]
@jasnell can you add this to the document please?

**Comment by evanp** [comment]
I'm taking this one over.

**Comment by jasnell** [comment]
Thank you @evanp ... I simply have not had the free cycles to dedicate to this.
